frontend/src/pages/patients/PatientsPage.tsx

```diff
@@ -65,8 +74,111 @@ function patientAliasFilterWhereVariables(value: string): PatientAliasWhere[] {
   }
 }
 
-export const PatientsPage: React.FunctionComponent = (props) => {
+function addCDashToCMOId(cmoId: string): string {
+  return cmoId.length === 6 ? `C-${cmoId}` : cmoId;
+}
+
+export default function PatientsPage({ setUserEmail }: { setUserEmail: any }) {
   const params = useParams();
+  const childRef = useRef<any>(null);
+
+  const [phiEnabled, setPhiEnabled] = useState(false);
+  const [patientIdsTriplets, setPatientIdsTriplets] = useState<
+    PatientIdsTriplet[]
+  >([]);
+  const [patientsListColumns, setPatientsListColumns] =
+    useState(PatientsListColumns);
+  const [popupOpen, setPopupOpen] = useState(false);
+  const [showAlertModal, setShowAlertModal] = useState(false);
+
+  useEffect(() => {
+    function handleLogin(event: any) {
+      if (event.origin !== "http://localhost:4001") return;
+      setUserEmail(event.data);
+      setPopupOpen(false);
+      childRef.current.handleSearch();
+    }
+
+    if (popupOpen) {
+      window.addEventListener("message", handleLogin);
```

Comment:
you can probably set up listener globally once and not worry about removing it like this.

Reply:
completely trivial!

Reply:
@alisman After more testing and research, I've decided to mostly keep this code as-is. Native DOM methods like `addEventListener` don't stay up to date with React's state variable changes ([a bit more info here](https://stackoverflow.com/questions/67244161/react-js-state-not-updating-in-event-listener-in-useeffect-hook)). Because the `handleLogin` function relies on certain React state variables, if we add this listener at the beginning of the page load, `handleLogin` won't work correctly. Happy to discuss this with you live.